### PR TITLE
Added hasOwnProperty test for when object has a similarly named property.

### PR DESCRIPTION
**Pull request by rw251**
hasOwnProperty is not a reserved word, so you can have objects with properties called this.  If that is the case then to list the properties of an object you must use the syntax:

({}).hasOwnProperty.call(obj, prop)

### CODE REVIEW

File: app/objects.js [hunk 1]
```diff
@@ -12,6 +12,10 @@ define(function() {
 
     iterate : function(obj) {
 
+    },
+
+    iterate2 : function(obj) {
```
Comment (rmurphey) [inline app/objects.js]: How about `iterateOwn` instead of `iterate2`?

**Comment by jmeas** [comment]
Thanks for the PR @rw251! This seems like such an edge case that the majority of developers would not run into, so I'm not sure if it's worth putting a test in there for it. I think – for now – I'm gonna close the issue.

If you feel strongly about this test, **do** leave a comment here and we can definitely keep the conversation going.

Thanks again!
